ui/apps/platform/src/Containers/NetworkGraph/cidr/CidrBlockSideBar.tsx

```diff
@@ -1,22 +1,167 @@
-import React from 'react';
-import { Badge, Flex, FlexItem, Text, TextContent, TextVariants } from '@patternfly/react-core';
+import React, { ReactElement } from 'react';
+import {
+    Divider,
+    Flex,
+    FlexItem,
+    Stack,
+    StackItem,
+    Text,
+    TextContent,
+    TextVariants,
+    Toolbar,
+    ToolbarContent,
+    ToolbarItem,
+} from '@patternfly/react-core';
+import { EdgeModel, NodeModel } from '@patternfly/react-topology';
+
+import { getNodeById } from '../utils/networkGraphUtils';
+import { getAllUniquePorts, getNumFlows } from '../utils/flowUtils';
+import { AdvancedFlowsFilterType } from '../common/AdvancedFlowsFilter/types';
+import { Flow } from '../types';
+
+import { CidrBlockIcon } from '../common/NetworkGraphIcons';
+import AdvancedFlowsFilter, {
+    defaultAdvancedFlowsFilters,
+} from '../common/AdvancedFlowsFilter/AdvancedFlowsFilter';
+import EntityNameSearchInput from '../common/EntityNameSearchInput';
+import FlowsTable from '../common/FlowsTable';
+import FlowsTableHeaderText from '../common/FlowsTableHeaderText';
+import FlowsBulkActions from '../common/FlowsBulkActions';
+
+type CidrBlockSideBarProps = {
+    id: string;
+    nodes: NodeModel[];
+    edges: EdgeModel[];
+};
+
+const flows: Flow[] = [
+    {
+        id: 'Deployment 1-naples-Ingress-Many-TCP',
+        type: 'Deployment',
+        entity: 'Deployment 1',
+        namespace: 'naples',
+        direction: 'Ingress',
+        port: '9000',
+        protocol: 'TCP',
+        isAnomalous: true,
+        children: [],
+    },
+    {
+        id: 'Deployment 2-naples-Ingress-Many-UDP',
+        type: 'Deployment',
+        entity: 'Deployment 2',
+        namespace: 'naples',
+        direction: 'Ingress',
+        port: '8080',
+        protocol: 'UDP',
+        isAnomalous: false,
+        children: [],
+    },
+    {
+        id: 'Deployment 3-naples-Egress-7777-UDP',
+        type: 'Deployment',
+        entity: 'Deployment 3',
+        namespace: 'naples',
+        direction: 'Egress',
+        port: '7777',
+        protocol: 'UDP',
+        isAnomalous: false,
+        children: [],
+    },
+];
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function CidrBlockSideBar({ id, nodes, edges }: CidrBlockSideBarProps): ReactElement {
+    // component state
+    const [entityNameFilter, setEntityNameFilter] = React.useState<string>('');
+    const [advancedFilters, setAdvancedFilters] = React.useState<AdvancedFlowsFilterType>(
+        defaultAdvancedFlowsFilters
+    );
+    const initialExpandedRows = flows
+        .filter((row) => row.children && !!row.children.length)
+        .map((row) => row.id); // Default to all expanded
+    const [expandedRows, setExpandedRows] = React.useState<string[]>(initialExpandedRows);
+    const [selectedRows, setSelectedRows] = React.useState<string[]>([]);
+
+    // derived data
+    const cidrBlockNode = getNodeById(nodes, id);
```

Comment:
Are you going to need all the `nodes` at a later iteration of this component? (similar question about `edges` )

If not, it may be worth having the parent compute this and past it in.

Reply:
Good point. I think for now we'll keep it and decide if we need to refactor things in a future PR